src/Sankey.js

```diff
@@ -0,0 +1,201 @@
+import {sankey, sankeyLinkHorizontal} from "d3-sankey";
+
+import {accessor, assign, configPrep, constant, elem} from "d3plus-common";
+import {Path, Rect} from "d3plus-shape";
+import {dataLoad as load, Viz} from "d3plus-viz";
+
+/**
+    @class Rings
+    @extends external:Viz
+    @desc Creates a sankey visualization based on a defined set of XXXXX. [Click here](xxxxxx) for help getting started using the Sankey class.
+*/
+export default class Sankey extends Viz {
+
+  /**
+      @memberof Sankey
+      @desc Invoked when creating a new class instance, and sets any default parameters.
+      @private
+  */
+  constructor() {
+    super();
+    this._nodeId = false;
+    this._links = accessor("links");
+    this._noDataMessage = false;
+    this._nodes = accessor("nodes");
+    this._nodeWidth = 30;
+    this._sankey = sankey();
+    this._shapeConfig = assign(this._shapeConfig, {
+      labelConfig: {
+        duration: 0,
+        fontMin: 1,
+        fontResize: true,
+        labelPadding: 0,
+        textAnchor: "middle",
+        verticalAlign: "middle"
+      },
+      Path: {
+        fill: "none",
+        label: false,
+        stroke: "#dbdbdb",
+        strokeWidth: 20,
+        strokeOpacity: 0.4
+      }
+    });
+  }
+
+  /**
+      Extends the draw behavior of the abstract Viz class.
+      @private
+  */
+  _draw(callback) {
+    super._draw(callback);
+
+    const height = this._height - this._margin.top - this._margin.bottom,
+          width = this._width - this._margin.left - this._margin.right;
+
+    const path = sankeyLinkHorizontal();
+
+    const nodes = this._nodes;
+
+    const nodeLookup = this._nodeLookup = this._nodes.reduce((obj, d, key) => {
+      obj[d[this._nodeId || "id"]] = key;
+      return obj;
+    }, {});
+
```

Comment:
Create the linkLookup the way Network does it, so that we can then access the lookup in a custom hover function so that we can highlight the links while hovering over nodes: https://github.com/d3plus/d3plus-network/blob/master/src/Network.js#L295-L301